### PR TITLE
Adding a second solution with a "for" loop

### PR DESCRIPTION
**Pull request by codingk8**
+ adapted some texts to make the whole thing coherent
+ added some Useful links at the end of the article

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

**Comment by codingk8** [comment]
Please let me know if I could have improved something in this PR.

File: guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md]:
```suggestion
The `this` keyword gives us access to the object we are calling myMap on.
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md]:
```suggestion
### Solution 1 - Solve this challenge using `this` and the forEach method
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md]:
```suggestion
### Solution 2 - Solve this challenge using a "for" loop and `this`
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md]:
```suggestion
[this. JavaScript MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this)<br/>
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md]:
```suggestion
[this. JavaScript W3Schools](https://www.w3schools.com/js/js_this.asp)   
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md]:
```suggestion
[for loop MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for)   
```

File: guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md
Comment (thecodingaviator) [inline guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype/index.md]:
```suggestion
[Array.prototype MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/prototype)   
```

**Review by thecodingaviator — CHANGES_REQUESTED** [review]
Please check and make the required changes

**Comment by codingk8** [comment]
All done I guess. Thanks!

**Comment by thecodingaviator** [comment]
I had used backticks for marking code, which you've changed to ", please set it back to `

**Comment by codingk8** [comment]
Right, sorry about that. Should be OK now. Thank!

**Comment by codingk8** [comment]
Sorry to bother you. It's been 10 days and it's still open. Is there a pb? Is it normal ? Thanks for letting me know 😃

**Comment by thecodingaviator** [comment]
Yup, such waiting times are normal since we are still dealing with a lot of Pull Requests from October last  year. We're hierarchically going through the PRs from oldest to latest

**Comment by codingk8** [comment]
OK fine. It's good to know. So I just wait until it's merged.

**Comment by RandellDawson** [comment]
@codingk8 There is only one PR ahead of yours in terms of the file you changed, but it affects many other files.  This means, we are having to review many other earlier PRs (about 20) which affect files on the other PR first.  Probably 1-2 weeks out is my guess.

**Comment by thecodingaviator** [comment]
@RandellDawson Do we want the solutions to be named solution 1 and solution 2 or just solutions or intermediate and advanced, like how we have it in the other JS challenges

**Comment by RandellDawson** [comment]
@thecodingaviator Personally, it does not matter since we have no true policy in the docs about this.  As you know, issue #35453 hopes to standardize and create a formal policy around these types of articles.

**Review by thecodingaviator — APPROVED** [review]
LGTM! :+1:

**Comment by thecodingaviator** [comment]
Thanks for contributing to the freeCodeCamp repository. Congratulations on your first PR! 🎊 

Look forward to more contributions from you! ✈️

**Pull request merged**
